src/movies.js

```diff
@@ -1,25 +1,100 @@
+
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const allDirectors = moviesArray.map((movie) => {
+        return movie.director;
+    })
+    
+    return allDirectors;
+};
+
+/* --------------------------------BONUS
+
+    let uniqueDirectors = [];
+        moviesArray.forEach((element) => {
+            if (!uniqueDirectors.includes(element)) {
+                uniqueDirectors.push(element);
+            }
+}); */
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const justSpielberg = moviesArray.filter((movie) => {
+        return movie.director === 'Steven Spielberg';
+    })
+    const justSpielbergDrama = justSpielberg.filter((movie) => {
+        return movie.genre.includes('Drama');
+    })
+
+    return justSpielbergDrama.length;
```

Comment:
Once again, everything is correct, but you could return everything at once. Using the methods you used and the `&&` operator, like this:
```
return moviesArray.filter(
    (movies) => movies.director === 'Steven Spielberg' && movies.genre.includes('Drama')
  ).length;
```